js/rendererjs/statusbar.js

```diff
@@ -1,14 +1,59 @@
 import React, { PropTypes } from 'react'
 
-const redColor = '#E0000B'
-const greenColor = '#00CBA0'
-const yellowColor = '#E7D414'
+// -- helper functions --
 
-const syncStyle = {
-	color: redColor,
+// currentEstimatedHeight returns the estimated block height for the current time.
+const currentEstimatedHeight = () => {
+	const knownBlockHeight = 98149
+	const knownBlockTime = new Date('March 31, 2017 23:30:30 UTC')
+	const blockTime = 9 //minutes
+	const diffMinutes = Math.abs(new Date() - knownBlockTime) / 1000 / 60 / 60
+
+	const estimatedHeight = knownBlockHeight + (diffMinutes / blockTime)
+
+	return Math.floor(estimatedHeight + 0.5) // round to the nearest block
 }
 
+// estimatedProgress returns the estimated sync progress given the current
+// blockheight, as a number from 0 -> 100
+const estimatedProgress = (currentHeight) =>
+	currentHeight / currentEstimatedHeight() * 100
```

Comment:
In siac we explicitly avoid displaying 100% progress, instead capping at 99%. Should probably do the same here.